An error occurred while compiling to a 32-bit device

```log
SEVERE: STDERR:
SEVERE: Compiling llrt_json v0.6.2-beta (https://github.com/awslabs/llrt#e2999601)
SEVERE:    Compiling zstd-safe v7.2.4
SEVERE:    Compiling zstd-sys v2.0.15+zstd.1.5.7
SEVERE:    Compiling brotlic-sys v0.2.2
SEVERE:    Compiling llrt_timers v0.6.2-beta (https://github.com/awslabs/llrt#e2999601)
SEVERE:    Compiling llrt_async_hooks v0.6.2-beta (https://github.com/awslabs/llrt#e2999601)
SEVERE:    Compiling llrt_buffer v0.6.2-beta (https://github.com/awslabs/llrt#e2999601)
SEVERE:    Compiling llrt_events v0.6.2-beta (https://github.com/awslabs/llrt#e2999601)
SEVERE:    Compiling llrt_exceptions v0.6.2-beta (https://github.com/awslabs/llrt#e2999601)
SEVERE:    Compiling llrt_numbers v0.6.2-beta (https://github.com/awslabs/llrt#e2999601)
SEVERE:    Compiling brotlic v0.8.2
SEVERE:    Compiling llrt_dns_cache v0.6.2-beta (https://github.com/awslabs/llrt#e2999601)
SEVERE: error[E0610]: `u64` is a primitive type and therefore doesn't have fields
SEVERE:    --> /Users/iota9star/.cargo/git/checkouts/llrt-0a7fdb298878dfd0/e299960/modules/llrt_async_hooks/src/lib.rs:214:46
SEVERE:     |
SEVERE: 214 |                 .map(|v| unsafe { v.as_raw().u.ptr } as usize)
SEVERE:     |                                              ^
SEVERE: 
SEVERE: error[E0610]: `u64` is a primitive type and therefore doesn't have fields
SEVERE:    --> /Users/iota9star/.cargo/git/checkouts/llrt-0a7fdb298878dfd0/e299960/libs/llrt_json/src/stringify.rs:320:66
SEVERE:     |
SEVERE: 320 |     let parent_ptr = unsafe { parent.unwrap_unchecked().as_raw().u.ptr as usize };
SEVERE:     |                                                                  ^
SEVERE: 
SEVERE: error[E0610]: `u64` is a primitive type and therefore doesn't have fields
SEVERE:    --> /Users/iota9star/.cargo/git/checkouts/llrt-0a7fdb298878dfd0/e299960/libs/llrt_json/src/stringify.rs:321:47
SEVERE:     |
SEVERE: 321 |     let current_ptr = unsafe { value.as_raw().u.ptr as usize };
SEVERE:     |                                               ^
SEVERE: 
SEVERE: error[E0610]: `u64` is a primitive type and therefore doesn't have fields
SEVERE:   --> /Users/iota9star/.cargo/git/checkouts/llrt-0a7fdb298878dfd0/e299960/modules/llrt_timers/src/lib.rs:82:36
SEVERE:    |
SEVERE: 82 |     let uid = unsafe { cb.as_raw().u.ptr } as usize;
SEVERE:    |                                    ^
SEVERE: 
SEVERE: error[E0610]: `u64` is a primitive type and therefore doesn't have fields
SEVERE:    --> /Users/iota9star/.cargo/git/checkouts/llrt-0a7fdb298878dfd0/e299960/modules/llrt_timers/src/lib.rs:100:36
SEVERE:     |
SEVERE: 100 |     let uid = unsafe { cb.as_raw().u.ptr } as usize;
SEVERE:     |                                    ^
SEVERE: 
SEVERE: error[E0610]: `u64` is a primitive type and therefore doesn't have fields
SEVERE:    --> /Users/iota9star/.cargo/git/checkouts/llrt-0a7fdb298878dfd0/e299960/modules/llrt_timers/src/lib.rs:375:60
SEVERE:     |
SEVERE: 375 |                 let uid: usize = unsafe { timeout.as_raw().u.ptr } as usize;
SEVERE:     |                                                            ^
SEVERE: 
SEVERE: For more information about this error, try `rustc --explain E0610`.
SEVERE: error[E0610]: `u64` is a primitive type and therefore doesn't have fields
SEVERE:    --> /Users/iota9star/.cargo/git/checkouts/llrt-0a7fdb298878dfd0/e299960/modules/llrt_async_hooks/src/lib.rs:218:46
SEVERE:     |
SEVERE: 218 |                 .map(|v| unsafe { v.as_raw().u.ptr } as usize);
SEVERE:     |                                              ^
SEVERE: 
SEVERE: error: could not compile `llrt_timers` (lib) due to 3 previous errors
SEVERE: warning: build failed, waiting for other jobs to finish...
SEVERE: error: could not compile `llrt_json` (lib) due to 2 previous errors
SEVERE: error: could not compile `llrt_async_hooks` (lib) due to 2 previous errors

```

--- a/libs/llrt_json/src/stringify.rs
+++ b/libs/llrt_json/src/stringify.rs
@@ -3,7 +3,8 @@
 use std::{collections::HashSet, rc::Rc};
 
 use rquickjs::{
-    atom::PredefinedAtom, function::This, Ctx, Exception, Function, Object, Result, Type, Value,
+    atom::PredefinedAtom, function::This, qjs, Ctx, Exception, Function, Object, Result, Type,
+    Value,
 };
 
 use crate::escape::escape_json_string;
@@ -317,8 +318,8 @@ fn detect_circular_reference(
     ancestors: &mut Vec<(usize, Rc<str>)>,
     itoa_buffer: &mut itoa::Buffer,
 ) -> Result<()> {
-    let parent_ptr = unsafe { parent.unwrap_unchecked().as_raw().u.ptr as usize };
-    let current_ptr = unsafe { value.as_raw().u.ptr as usize };
+    let parent_ptr = unsafe { qjs::JS_VALUE_GET_PTR(parent.unwrap_unchecked().as_raw()) as usize };
+    let current_ptr = unsafe { qjs::JS_VALUE_GET_PTR(value.as_raw()) as usize };
 
     while !ancestors.is_empty()
         && match ancestors.last() {

--- a/llrt_core/src/modules/require/mod.rs
+++ b/llrt_core/src/modules/require/mod.rs
@@ -138,7 +138,7 @@ pub fn require(ctx: Ctx<'_>, specifier: String) -> Result<Value<'_>> {
     let mut executing_timers = Vec::new();
 
     // SAFETY: Since it checks in advance whether it is an Object type, we can always get a pointer to the object.
-    let uid = unsafe { obj.as_object().unwrap().as_raw().u.ptr } as usize;
+    let uid = unsafe { qjs::JS_VALUE_GET_PTR(obj.as_object().unwrap().as_raw()) } as usize;
     register_finalization_registry(&ctx, obj.clone().into_value(), uid)?;
     invoke_async_hook(&ctx, HookType::Init, ProviderType::TimerWrap, uid)?;
 

--- a/modules/llrt_async_hooks/src/lib.rs
+++ b/modules/llrt_async_hooks/src/lib.rs
@@ -11,6 +11,7 @@ use rquickjs::{
     module::{Declarations, Exports, ModuleDef},
     prelude::Func,
     promise::PromiseHookType,
+    qjs,
     runtime::PromiseHook,
     Ctx, Function, JsLifetime, Object, Result, Value,
 };
@@ -211,11 +212,11 @@ pub fn promise_hook_tracker() -> PromiseHook {
             // SAFETY: Since it checks in advance whether it is an Object type, we can always get a pointer to the object.
             let object = promise
                 .as_object()
-                .map(|v| unsafe { v.as_raw().u.ptr } as usize)
+                .map(|v| unsafe { qjs::JS_VALUE_GET_PTR(v.as_raw()) } as usize)
                 .unwrap();
             let parent = parent
                 .as_object()
-                .map(|v| unsafe { v.as_raw().u.ptr } as usize);
+                .map(|v| unsafe { qjs::JS_VALUE_GET_PTR(v.as_raw()) } as usize);
 
             if type_ == PromiseHookType::Init {
                 let _ = register_finalization_registry(&ctx, promise, object);

--- a/modules/llrt_dns/src/lib.rs
+++ b/modules/llrt_dns/src/lib.rs
@@ -11,7 +11,7 @@ use llrt_utils::{
 use rquickjs::{
     module::{Declarations, Exports, ModuleDef},
     prelude::{Func, Rest},
-    Ctx, Error, Exception, Function, IntoJs, Null, Result, Value,
+    qjs, Ctx, Error, Exception, Function, IntoJs, Null, Result, Value,
 };
 
 fn lookup<'js>(ctx: Ctx<'js>, hostname: String, args: Rest<Value<'js>>) -> Result<()> {
@@ -22,7 +22,7 @@ fn lookup<'js>(ctx: Ctx<'js>, hostname: String, args: Rest<Value<'js>>) -> Resul
         .or_throw_msg(&ctx, "Callback parameter is not a function")?;
 
     // SAFETY: Since it checks in advance whether it is an Function type, we can always get a pointer to the Function.
-    let uid = unsafe { cb.as_raw().u.ptr } as usize;
+    let uid = unsafe { qjs::JS_VALUE_GET_PTR(cb.as_raw()) } as usize;
     register_finalization_registry(&ctx, cb.clone().into_value(), uid)?;
     invoke_async_hook(&ctx, HookType::Init, ProviderType::GetAddrInfoReqWrap, uid)?;
 

--- a/modules/llrt_timers/src/lib.rs
+++ b/modules/llrt_timers/src/lib.rs
@@ -79,7 +79,7 @@ impl Default for Timeout {
 
 fn queue_microtask<'js>(_ctx: Ctx<'js>, cb: Function<'js>) -> Result<()> {
     // SAFETY: Since it checks in advance whether it is an Function type, we can always get a pointer to the Function.
-    let uid = unsafe { cb.as_raw().u.ptr } as usize;
+    let uid = unsafe { qjs::JS_VALUE_GET_PTR(cb.as_raw()) } as usize;
     register_finalization_registry(&_ctx, cb.clone().into_value(), uid)?;
     invoke_async_hook(&_ctx, HookType::Init, ProviderType::Microtask, uid)?;
     // NOTE: Defer simply registers a task in a microtask queue
@@ -97,7 +97,7 @@ pub fn set_timeout_interval<'js>(
     provider_type: ProviderType,
 ) -> Result<usize> {
     // SAFETY: Since it checks in advance whether it is an Function type, we can always get a pointer to the Function.
-    let uid = unsafe { cb.as_raw().u.ptr } as usize;
+    let uid = unsafe { qjs::JS_VALUE_GET_PTR(cb.as_raw()) } as usize;
 
     // NOTE: https://noncodersuccess.medium.com/understanding-setimmediate-vs-settimeout-in-node-js-6a3ef8fc02d4
     // If `setImmediate(fn)` and `setTimeout(fn, 0) are queued at the exact same time,
@@ -372,7 +372,7 @@ pub fn poll_timers(
 
             if let Ok(timeout) = timeout.restore(&ctx2) {
                 // SAFETY: Since it checks in advance whether it is an Function type, we can always get a pointer to the Function.
-                let uid: usize = unsafe { timeout.as_raw().u.ptr } as usize;
+                let uid: usize = unsafe { qjs::JS_VALUE_GET_PTR(timeout.as_raw()) } as usize;
 
                 invoke_async_hook(&ctx2, HookType::Before, ProviderType::None, uid)?;
 

